pkg/iac/terraform/state/backend/http_reader.go

```diff
@@ -46,7 +46,11 @@ func (h *HTTPBackend) Read(p []byte) (n int, err error) {
 			return 0, errors.Errorf("error requesting HTTP(s) backend state: status code: %d", res.StatusCode)
 		}
 	}
-	return h.reader.Read(p)
+	read, err := h.reader.Read(p)
+	if err != nil {
+		return 0, err
+	}
+	return read, nil
```

Comment:
What's the point of this change ? `h.reader.Read(p)` already returns an error so this shouldn't have any effect at all, should it?

Reply:
It's work because i return 0 in the case err != nil, but it erase all error to print "no state".
So i'll rework this one.